src/Data/Authenticator.php

```diff
@@ -0,0 +1,97 @@
+<?php
+namespace App\Data;
+
+use App\Data\Entity\Login;
+use App\Data\Entity\User;
+use App\Data\Mapper\LoginMapper;
+use App\Data\Mapper\UserMapper;
+use Carbon\Carbon;
+use Spark\Auth\AbstractAuthenticator;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class Authenticator extends AbstractAuthenticator
+{
+    /**
+     * @var Login
+     */
+    protected $login;
+
+    /**
+     * @var LoginMapper
+     */
+    protected $mapper;
+
+    /**
+     * @var UserMapper
+     */
+    protected $userMapper;
+
+    protected $secret;
+
+    public function __construct(LoginMapper $mapper, UserMapper $userMapper)
+    {
+        $this->mapper = $mapper;
+        $this->userMapper = $userMapper;
+        $this->secret = $_ENV['JWT_SECRET'];
```

Comment:
We can probably add a `EnvironmentBag` class that acts like `SessionBag`. Testable and easy to implement.